### PR TITLE
T17: Generate string permutations

### PR DESCRIPTION
**Pull request by Ashish02112001**

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -1,7 +1,45 @@
...
+      static string[] Permute1 (string word) {
+         int count = Factorial (word.Length);
+         string[] result = new string[count];
+         if (word.Length == 1) { result[0] = word; return result; }
+         for (int i = 0; i < word.Length; i++) {
+            char start = word[i];
+            string remainingStr = word.Remove (i, 1);
+            var remWords = Permute1 (remainingStr);
+            int tCount = Factorial (remainingStr.Length);
+            for (int k = 0; k < tCount; k++) result[k + i * tCount] = start + remWords[k];
```
Comment (asfar-nazar) [inline Program.cs:38]: For clarity, better use result as a list<string> instead of string array

File: Program.cs [hunk 2]
```diff
@@ -1,7 +1,45 @@
...
    internal class Program {
+      #region Methods ---------------------------------------------
+      /// <summary>This method gets a string and gives the permutations of the string</summary>
+      /// <param name="args">arguments</param>
       static void Main (string[] args) {
-         Console.WriteLine ("Hello, World!");
+         int count = 1;
+         string word = "RED";
+         Console.WriteLine ($"Permutation of the word {word}");
+         foreach (string str in Permute1 (word)) { Console.WriteLine ($"{count,4}. {str}"); count++; }
```
Comment (asfar-nazar) [inline Program.cs:19]:
```suggestion
         foreach (string str in Permute1 (word)) { Console.WriteLine ($"{count, 4}. {str}"); count++; }
```

File: Program.cs [hunk 3]
```diff
@@ -1,7 +1,45 @@
...
+   #region Program ------------------------------------------------------------------------------
+   /// <summary>String permutations generator</summary>
    internal class Program {
+      #region Methods ---------------------------------------------
+      /// <summary>This method gets a string and gives the permutations of the string</summary>
+      /// <param name="args">arguments</param>
       static void Main (string[] args) {
-         Console.WriteLine ("Hello, World!");
+         int count = 1;
+         string word = "RED";
```
Comment (asfar-nazar) [inline Program.cs:17]: Get word as input from the user

**Review by asfar-nazar — CHANGES_REQUESTED** [review]

File: Training.csproj [hunk 4]
```diff
@@ -4,7 +4,6 @@
     <OutputType>Exe</OutputType>
     <TargetFramework>net7.0</TargetFramework>
     <ImplicitUsings>enable</ImplicitUsings>
-    <Nullable>enable</Nullable>
```
[the remark this answers is not in the document]
Comment (Ashish02112001) [reply Training.csproj:7]: I have got null error at some line but don't know where. So i have removed it na.

File: Program.cs (hunk 2 above)
Comment (Ashish02112001) [reply Program.cs:19]: I don't get it na
Can you brief it?

File: Training.csproj (hunk 4 above)
[the remark this answers is not in the document]
Comment (asfar-nazar) [reply Training.csproj:7]: Undo this and use the null coalescing operator when assigning 'word' variable

**Review by asfar-nazar — CHANGES_REQUESTED** [review]

**Pull request closed without merging**